controllers/controller/workspacerouting/predictes.go

```diff
@@ -0,0 +1,58 @@
+//
+// Copyright (c) 2019-2020 Red Hat, Inc.
+// This program and the accompanying materials are made
+// available under the terms of the Eclipse Public License 2.0
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+//
+// SPDX-License-Identifier: EPL-2.0
+//
+// Contributors:
+//   Red Hat, Inc. - initial API and implementation
+//
+
+package workspacerouting
+
+import (
+	"errors"
+
+	controllerv1alpha1 "github.com/devfile/devworkspace-operator/apis/controller/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+var routingPredicates = predicate.Funcs{
+	CreateFunc: func(ev event.CreateEvent) bool {
+		obj, ok := ev.Object.(*controllerv1alpha1.WorkspaceRouting)
+		if !ok {
+			return true
+		}
+		if _, err := getSolverForRoutingClass(obj.Spec.RoutingClass); errors.Is(err, externalRoutingError) {
+			return false
+		}
+		return true
+	},
+	DeleteFunc: func(_ event.DeleteEvent) bool {
+		// Return true to ensure finalizers are removed
```

Comment:
That's also invoked when Service or Ingress is removed, right?
If so - this comment is a bit misleading, so - we should process:
- all Workspace Routing independent from their ingresses routes 
- all other routing-related objects (Service, Ingress, .. ) to recreate object if needed.

Reply:
Updated the comment in #219